### DISCUSSION TITLE
Convert Markdown links to <Link> components

### DISCUSSION THREAD

**Discussion by lpsinger** [remix-run/remix#6048] [Q&A] [answered]
In `.md` and `.mdx` routes, links that are written in Markdown syntax like `[link text](url)` are rendered as HTML `<a>` elements, so they do not participate in Remix's nice client-side routing. It would be very nice if links in Markdown were rendered as `<Link>` components.

**Comment by sergiodxa** [comment, marked as answer]
You could use https://github.com/sergiodxa/remix-utils#delegate-anchors-to-remix to make them work like Link

**Reply by lpsinger** [reply]
How would I deploy this for .md routes?

**Reply by sergiodxa** [reply]
Create a pathless layout route that wraps your `.md` routes and put it there.

**Reply by lpsinger** [reply]
`useDelegatedAnchors` requires a ref, but the `<Outlet/>` component does not have a `ref` property. How would I apply `useDelegatedAnchors` to child routes?

**Reply by sergiodxa** [reply]
wrap <Outlet /> in a div with a `ref`
```tsx
let ref = useRef<HTMLDivElement>(null)
useDelegatedAnchors(ref)
return <div ref={ref}><Outlet /></div>
```
or use [PrefetchPageAnchors](https://github.com/sergiodxa/remix-utils#delegate-anchors-to-remix)
```tsx
return <PrefetchPageAnchors><Outlet /></PrefetchPageAnchors>
```
It uses `useDelegatedAnchors` for you wrapping the content in a div, and also supports prefetching links.

**Reply by lpsinger** [reply]
This solution isn't working in Remix 1.15.0. @sergiodxa, would you please merge https://github.com/sergiodxa/remix-utils/pull/187?

**Comment by lpsinger** [comment]
#7502 would permit a much simpler solution.
